### ISSUE TITLE
Make update_environment_build() function overridable via traitlets

### ISSUE THREAD

**Issue by aisis-julian** [Quansight/conda-store#421]
Our team is looking for a solution to sync conda environments onto EFS in a “background” job so that we can keep the performance benefits of building environments with fast block storage but be able to also populate the conda environments on EFS for use in other parts of our system like papermill/voila. Right now we’ve implemented a custom storage class for conda-store and implemented the fset() method to unpack conda pack archives into a directory on EFS before calling the super method, this works well except for the case of rolling back environment builds since the update_environment_build() method always uses the same build_path and environment_path when updating the symlinks.

**Comment by jonzeper** [comment]
Wondering if it would make sense to implement a configurable hook instead of overriding the whole function. If we don't have to reimplement the whole method, makes future upgrades safer.
e.g:

```python
# app.py
# Config
post_update_environment_build_hook = Callable(...)

def update_environment_build():
  # ... all the existing logic ...
  if self.post_update_environment_build_hook:
    self.post_update_environment_build_hook(env, build)
```

**Comment by costrouc** [comment]
Closing since resolved in PR #425
